server/cshr/api/permission.py

```diff
@@ -0,0 +1,59 @@
+from rest_framework import permissions
+from django.core.exceptions import PermissionDenied
+from rest_framework.request import Request
+from server.cshr.services.users import get_user_type_by_id
+
+from rest_framework.views import APIView
+
+
+class UserIsAuthenticated(permissions.BasePermission):
+    """
+    logged in permission
+    """
+
+    def has_permission(self, request: Request, view: APIView) -> bool:
+        if request.user.is_authenticated:
+            return True
+        raise PermissionDenied
+
+
+class IsAdmin(permissions.BasePermission):
+    """
+    admin permission
+    """
+
+    def has_permission(self, request: Request, view: APIView) -> bool:
+        if request.user.is_authenticated:
+            userType = get_user_type_by_id(request.user.id)
+            if userType == "Admin":
```

Comment:
When we use some Emun and make a condition on it we have to use its choice field instead of a hard-coded type.
```suggestion
            if userType == UserType.Admin:
```